ingress/src/test/java/com/redhat/service/bridge/ingress/api/IngressAPITest.java

```diff
@@ -43,47 +44,51 @@ public static void setup() {
 
     @Test
     public void testSendCloudEvent() throws JsonProcessingException {
-        ingressService.deploy("topicName");
-
-        given()
-                .filter(new ResponseLoggingFilter())
-                .contentType(ContentType.JSON)
-                .when()
-                .body(CloudEventUtils.encode(TestUtils.buildTestCloudEvent()))
-                .post("/ingress/events/topicName")
-                .then().statusCode(200);
-
+        doApiCallAfterDeploy(TestUtils.buildTestCloudEvent(), 200);
         verify(kafkaEventPublisher, times(1)).sendEvent(eq("topicName"), any(CloudEvent.class));
-        ingressService.undeploy("topicName");
     }
 
     @Test
-    public void testNonCloudEvent() throws JsonProcessingException {
-        ingressService.deploy("topicName");
-
-        given()
-                .filter(new ResponseLoggingFilter())
-                .contentType(ContentType.JSON)
-                .when()
-                .body("{\"key\": \"not a cloud event\"}")
-                .post("/ingress/events/topicName")
-                .then().statusCode(400);
+    public void testSendCloudEventWithBadRequestException() throws JsonProcessingException {
+        Mockito.doThrow(BadRequestException.class).when(kafkaEventPublisher).sendEvent(any(String.class), any(CloudEvent.class));
```

Comment:
@kostola I wonder if it's better to invoke the "real" class in this instance rather than the Mock via `thenCallRealMethod` on Mockito. In that case we're actually invoking the code that should throw the Exception if the user is trying to use reserved attributes, rather than just throwing an Exception for any invocation of the Mock. WDYT?

Reply:
Good idea @robpblake! I changed it as you suggested and it's working :+1: